### PR TITLE
feat: Returns all failover errors from pact calls

### PR DESCRIPTION
**Pull request by benkolera**
We have a request to present all of the errors that happen when we get an error that looks to be transient to a node (i.e we should failover to another node). Otherwise those errors just disappear, and there may be something helpful to the user in there.

### CODE REVIEW

File: frontend/src/Frontend/Network.hs [hunk 1]
```diff
@@ -762,35 +787,32 @@ performNetworkRequestCustom
...
-            [] -> doReq Nothing req
-
+                               then go (errs <> DList.singleton err) ns
+                               else pure $ mkResult errs (Left err)
+                Right res -> pure $ mkResult errs (Right res)
+            [] -> mkResult errs <$> doReq Nothing req
+          mkResult errs (Left e) = This (NEL.fromList $ DList.toList (errs <> pure e))
+          mkResult errs (Right res) = maybe (That res) (flip These res) . nonEmpty . DList.toList $ errs
+
+    doReq :: Maybe NodeInfo        -> NetworkRequest -> JSM (Either NetworkError (Maybe Gas, PactValue))
```
Comment (tomsmalley) [inline frontend/src/Frontend/Network.hs]: strange spacing

File: frontend/src/Frontend/Network.hs [hunk 2]
```diff
@@ -762,35 +787,32 @@ performNetworkRequestCustom
...
-                               then go ns
-                               else pure errRes
-                Right _ -> pure errRes
-            [] -> doReq Nothing req
-
+                               then go (errs <> DList.singleton err) ns
+                               else pure $ mkResult errs (Left err)
+                Right res -> pure $ mkResult errs (Right res)
+            [] -> mkResult errs <$> doReq Nothing req
+          mkResult errs (Left e) = This (NEL.fromList $ DList.toList (errs <> pure e))
```
Comment (tomsmalley) [inline frontend/src/Frontend/Network.hs]: I'm not convinced of the value of using DList given the length of the list is always going to be fairly short. I'd rather just keep the NonEmpty list and either just pay the append cost or cons everything and reverse it at the end. We can avoid the NEL.fromList that way too.

**Review by tomsmalley — COMMENTED** [review]
Looks good; just nitpicks

File: frontend/src/Frontend/Network.hs (hunk 2 above)
Comment (benkolera) [reply frontend/src/Frontend/Network.hs]: Yeah, that's fair. It's just a reflex when doing this sort of stuff. :)

File: frontend/src/Frontend/Network.hs (hunk 1 above)
Comment (benkolera) [reply frontend/src/Frontend/Network.hs]: Yup, not sure what happened there. Lol.

File: frontend/src/Frontend/Network.hs (hunk 2 above)
Comment (benkolera) [reply frontend/src/Frontend/Network.hs]: Fixed by d672530

File: frontend/src/Frontend/Network.hs (hunk 1 above)
Comment (benkolera) [reply frontend/src/Frontend/Network.hs]: Fixed by d672530

**Comment by benkolera** [comment]
OK, should be good for proper review now. Was tricky to test because I'm a noob and kept getting distracted with other things.

**Pull request merged**